### ISSUE TITLE
Plugin folder location

### ISSUE THREAD

**Issue by ScottIsAFool** [OpenLiveWriter/OpenLiveWriter#74]
Writer chooses where to look for its plugins in essentially two places: the Plugins folder within the install directory, and a registry setting that plugin developers could write to so they can tell Writer where their plugin is located. All good with WLW when installing to program files, but with OLW, this might present a problem. With Squirrel creating an app-x.x.x.x folder for each update, if any plugins are put in a Plugins folder within there, when the next update comes through, those plugins will have to be moved after every update. This is not ideal.

Suggestion: in %localappdata%\OpenLiveWriter, we have a Plugins folder in there where plugins get installed to, the Writer could would have to then be altered to check there instead (as well as??) the Plugins folder in its own directory.

Issue #19  would probably have to take this into consideration with whatever solution is arrived at for a plugin gallery.

**Comment by pekiZG** [comment]
Scott really good suggestion for %AppData% folder.
I proposed same solution for https://github.com/OpenLiveWriter/OpenLiveWriter/issues/29

**Comment by ScottIsAFool** [comment]
I wonder whether Squirrel can create that sort of folder. @paulcbetts any thoughts?

**Comment by pekiZG** [comment]
@ScottIsAFool Think about write privilegies (see https://github.com/OpenLiveWriter/OpenLiveWriter/issues/20).

**Comment by ScottIsAFool** [comment]
It should be able to write to its own folder. And that discussion is talking about ProgramData isn't it?

**Comment by pekiZG** [comment]
Yes, Scott you are right
/me messed up; was not thinking clearly.

**Comment by timheuer** [comment]
Why not use the registry mechanism for PluginAssemblies that exists today like WLW has?  I don't think forcing a specific install location for plugin authors is great, IMO.  Just have OLW smart enough to know where to look for all plugins that are registered with the app.

**Comment by ScottIsAFool** [comment]
@timheuer The problem is, what sets the PluginAssemblies value? Are we expecting users to go into the registry and set that? I'm not suggesting forcing plugin developers to have to use that, I'm more suggesting the default location of the Plugins folder is now incorrect, and it's *that* which will need changing.

**Comment by timheuer** [comment]
For the plugins that I created, I created an installer and my installer has always set that value...it was what the WLW plugin documentation told authors to do as well.  If a plugin is a single assembly, then maybe in the "add plugins" UI we could just have a drag/drop/upload mechanism that allows the client to bin-place it in the right area?

**Comment by ScottIsAFool** [comment]
And that's fine, but what's the "right area"? As I said, if it's the plugins folder in the OpenLiveWriter/app-x.x.x.x/ folder, that will become invalid as soon as the app gets updated, so it needs to be a fixed folder. That make sense?

**Comment by timheuer** [comment]
Yep that makes sense -- you need a 'place' no matter what.  Got it.  Shutting up now.  I wonder if this Squirrel infrastructure might get us in the way of some things versus a traditional installer.  Squirrel makes auto-updates easy, but there are also other .NET things that do that as well that would allow us to have the more traditional install location mechanics

**Comment by ScottIsAFool** [comment]
All depends on what route we want to go with the plugin gallery. I really like the suggestion of having it as a nuget style gallery, which would make installing plugins really easy (and do away with plugin installers! YAY :D). But even if we do go that way, we still need to know where to put them (even if the user doesn't need to know :))

**Comment by pekiZG** [comment]
I'm also in the favor of nuget style.
Someone also suggested to look the way VisualStudioCode handles plugins

**Comment by shanselman** [comment]
I'm sure @paulcbetts would have a thought about Plugins in a Squirrel world. We already have inherited the Squirrel doesn't like locked down machines that won't let folks into %appdata% but if we simply accept that (and as such, don't fix that issue) then we can put the plugins there.

**Comment by paulcbetts** [comment]
@shanselman Squirrel actually has a fallback for locked-down machines, it'll Just Work (by abusing ProgramData). 

> Suggestion: in %localappdata%\OpenLiveWriter, we have a Plugins folder in there where plugins get installed to, the Writer could would have to then be altered to check there instead (as well as??) the Plugins folder in its own directory.

This isn't a terrible idea. Also remember that Squirrel apps get a chance to run their own code on upgrade (https://github.com/Squirrel/Squirrel.Windows/blob/master/docs/squirrel-events.md), so you could keep moving that plugins folder to the latest version. 

> I wonder if this Squirrel infrastructure might get us in the way of some things versus a traditional installer. 

Squirrel can do everything you want it to (except for things that require UAC), you just have to sometimes think in a Different Way than you're perhaps used to, because Traditional Installers are this black box that you have no control over. Squirrel is just a library, developers get to choose when updates happen, what happens on upgrade, etc etc.

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
